routes/associations.go

```diff
@@ -0,0 +1,1036 @@
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
+
+	// Parse LastSeenAssociationID (BlockHash) from LastSeenAssociationIdHex (string).
+	var lastSeenAssociationID *lib.BlockHash
+	if queryType == AssociationQueryTypeQuery {
+		if requestData.LastSeenAssociationID != "" {
+			lastSeenAssociationIdBytes, err := hex.DecodeString(requestData.LastSeenAssociationID)
+			if err != nil {
+				return nil, errors.New("invalid LastSeenAssociationID provided")
+			}
+			lastSeenAssociationID = lib.NewBlockHash(lastSeenAssociationIdBytes)
+		}
+	} else if queryType == AssociationQueryTypeCount {
+		if requestData.LastSeenAssociationID != "" {
+			return nil, errors.New("unsupported LastSeenAssociationID param for count operation")
+		}
+	} else {
+		return nil, errors.New("invalid query type") // This can never happen.
+	}
+
+	// Validate SortDescending.
+	if queryType == AssociationQueryTypeCount && requestData.SortDescending {
+		return nil, errors.New("unsupported SortDescending param for count operation")
+	}
+
+	// Construct association query.
+	assocationQuery := &lib.UserAssociationQuery{
+		TransactorPKID:         transactorPKID,
+		TargetUserPKID:         targetUserPKID,
+		AppPKID:                appPKID,
+		AssociationType:        []byte(requestData.AssociationType),
+		AssociationTypePrefix:  []byte(requestData.AssociationTypePrefix),
+		AssociationValue:       []byte(requestData.AssociationValue),
+		AssociationValuePrefix: []byte(requestData.AssociationValuePrefix),
+		Limit:                  requestData.Limit,
+		LastSeenAssociationID:  lastSeenAssociationID,
+		SortDescending:         requestData.SortDescending,
+	}
+	return assocationQuery, nil
+}
+
+func (fes *APIServer) CreatePostAssociation(ww http.ResponseWriter, req *http.Request) {
+	// Decode request body.
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := CreatePostAssociationRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, "CreatePostAssociation: problem parsing request body")
+		return
+	}
+
+	// Create UTXO view.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddInternalServerError(ww, "CreatePostAssociation: problem getting UTXO view")
+		return
+	}
+
+	// Parse TransactorPublicKeyBytes from TransactorPublicKeyBase58Check.
+	if requestData.TransactorPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, "CreatePostAssociation: must provide a TransactorPublicKeyBase58Check")
+		return
+	}
+	transactorPublicKeyBytes, _, err := fes.GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check(
+		requestData.TransactorPublicKeyBase58Check,
+		utxoView,
+	)
+	if err != nil {
+		_AddInternalServerError(ww, "CreatePostAssociation: problem getting public key for the transactor")
+		return
+	}
+
+	// Parse PostHashBytes from PostHashHex.
+	if requestData.PostHashHex == "" {
```

Comment:
we probably want to validate that the post for this post hash hex really exists.

Reply:
We do this in core. LMK if I should duplicate.